buffer/src/main/java/io/netty5/buffer/api/internal/SingleComponentIterator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty5.buffer.api.internal;
+
+import io.netty5.buffer.api.ComponentIterator;
+import io.netty5.buffer.api.ComponentIterator.Next;
+import io.netty5.buffer.api.Resource;
+
+import java.lang.ref.Reference;
+
+public final class SingleComponentIterator<T extends Next> implements ComponentIterator<T> {
+    private final T singleComponent;
+
+    @SuppressWarnings("unchecked")
+    public SingleComponentIterator(Object singleComponent) {
```

Comment:
Why you cant use `T extends Next` ?

Reply:
Because the generic type at the call site would actually have stricter bounds, so we would need to cast there. Effectively, this moves the cast into the constructor, which I think makes the code look less busy, but we can do either way.

Reply:
its fine ... just wanted to know if I am missing anything